Dependency problem jakarta.mail-api/javax.mail

Build of Opencast 10.0 throws NoClassDefFoundError (com/sun/mail/util/MailLogger) exception on startup.

As of version 3.4 of the Apache Karaf CXF feature, javax.mail has been replaced by jakarta.mail-api version 1.6.x which seems to cause the problem or at least its the only reference to jakarta.mail-api I can find with grep in the build directory. https://github.com/apache/cxf/commit/6bc7f0cc09395effeaa57ee92e4b8c5dce5c00a6

```
karaf@root()> bundle:requirements 134 #opencast-kernel
osgi.wiring.package; (&(osgi.wiring.package=javax.mail)(version>=1.4.0)(!(version>=2.0.0))) resolved by: osgi.wiring.package; javax.mail 1.6.0 from jakarta.mail-api [38]

karaf@root()> bundle:find-class MailLogger #returns nothing
```

```
2021-06-17 19:23:34,228 | ERROR | (Log:186) - [org.osgi.service.cm.ManagedService, org.opencastproject.kernel.mail.SmtpService, id=396, bundle=134/mvn:org.opencastproject/opencast-kernel/10.0]: Unexpected problem updating configuration org.opencastproject.kernel.mail.SmtpService
java.lang.NoClassDefFoundError: com/sun/mail/util/MailLogger
	at javax.mail.Session.initLogger(Session.java:260) ~[?:?]
	at javax.mail.Session.<init>(Session.java:244) ~[?:?]
	at javax.mail.Session.getInstance(Session.java:298) ~[?:?]
	at org.opencastproject.kernel.mail.BaseSmtpService.getSession(BaseSmtpService.java:224) ~[?:?]
	at org.opencastproject.kernel.mail.BaseSmtpService.configure(BaseSmtpService.java:212) ~[?:?]
	at org.opencastproject.kernel.mail.SmtpService.updated(SmtpService.java:124) ~[?:?]
	at org.apache.felix.cm.impl.helper.ManagedServiceTracker.updated(ManagedServiceTracker.java:189) ~[!/:?]
	at org.apache.felix.cm.impl.helper.ManagedServiceTracker.updateService(ManagedServiceTracker.java:152) [!/:?]
	at org.apache.felix.cm.impl.helper.ManagedServiceTracker.provideConfiguration(ManagedServiceTracker.java:85) [!/:?]
	at org.apache.felix.cm.impl.ConfigurationManager$ManagedServiceUpdate.provide(ConfigurationManager.java:1118) [!/:?]
	at org.apache.felix.cm.impl.ConfigurationManager$ManagedServiceUpdate.run(ConfigurationManager.java:1074) [!/:?]
	at org.apache.felix.cm.impl.UpdateThread.run0(UpdateThread.java:138) [!/:?]
	at org.apache.felix.cm.impl.UpdateThread.run(UpdateThread.java:105) [!/:?]
	at java.lang.Thread.run(Thread.java:829) [?:?]
Caused by: java.lang.ClassNotFoundException: com.sun.mail.util.MailLogger not found by jakarta.mail-api [38]
	at org.apache.felix.framework.BundleWiringImpl.findClassOrResourceByDelegation(BundleWiringImpl.java:1639) ~[?:?]
	at org.apache.felix.framework.BundleWiringImpl.access$200(BundleWiringImpl.java:80) ~[?:?]
	at org.apache.felix.framework.BundleWiringImpl$BundleClassLoader.loadClass(BundleWiringImpl.java:2053) ~[?:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:522) ~[?:?]
	... 14 more

```

**Workaround or Solution**
Adding a 1.6 implementation of javax.mail resolves the problem.
`<bundle>mvn:com.sun.mail/jakarta.mail/1.6.7</bundle>`

**Environment Information**
 - OS: Fedora 33
 - Java: openjdk 11.0.11 2021-04-20
 - Opencast version 10.0

--- a/assemblies/karaf-features/src/main/feature/feature.xml
+++ b/assemblies/karaf-features/src/main/feature/feature.xml
@@ -48,6 +48,7 @@
     <bundle>mvn:com.google.guava/guava/${guava.version}</bundle>
     <bundle>mvn:com.google.guava/failureaccess/${failureaccess.version}</bundle>
     <bundle>mvn:com.google.code.gson/gson/${gson.version}</bundle>
+    <bundle>mvn:com.sun.mail/jakarta.mail/${jakarta.mail.version}</bundle>
     <bundle>mvn:commons-fileupload/commons-fileupload/${commons-fileupload.version}</bundle>
     <bundle>mvn:commons-io/commons-io/${commons-io.version}</bundle>
     <!-- FIXME: commons-lang 2.6 only needed for org.springframework.ldap/1.3.1.RELEASE -->

--- a/pom.xml
+++ b/pom.xml
@@ -35,6 +35,7 @@
     <httpcomponents-httpclient.version>4.5.12</httpcomponents-httpclient.version>
     <httpcomponents-httpcore.version>4.4.13</httpcomponents-httpcore.version>
     <jackson.version>2.12.3</jackson.version>
+    <jakarta.mail.version>1.6.7</jakarta.mail.version>
     <jersey.version>2.29.1</jersey.version>
     <jettison.version>1.4.1</jettison.version>
     <jdk.version>1.8</jdk.version>
@@ -982,9 +983,9 @@
         <version>1.2.1</version>
       </dependency>
       <dependency>
-        <groupId>javax.mail</groupId>
-        <artifactId>mail</artifactId>
-        <version>1.4.7</version>
+        <groupId>com.sun.mail</groupId>
+        <artifactId>jakarta.mail</artifactId>
+        <version>${jakarta.mail.version}</version>
       </dependency>
       <dependency>
         <groupId>javax.inject</groupId>

